Library/ViewModels/DiscoveryPageViewModel.swift

```diff
@@ -149,14 +149,18 @@ public final class DiscoveryPageViewModel: DiscoveryPageViewModelType, Discovery
 
     self.asyncReloadData = self.projects.take(first: 1).ignoreValues()
 
-    self.showEmptyState = paramsChanged
-      .takeWhen(paginatedProjects.filter { $0.isEmpty })
-      .map(emptyState(forParams:))
+    self.showEmptyState = Signal.combineLatest(paramsChanged, self.projectsAreLoading, paginatedProjects)
```

Comment:
Still just wondering about the `combineLatest` here. Remember that means that `self.showEmptyState` will emit when any of these three signals emit and also not emit until all three have emitted at least once.

Perhaps another way to think about it is - what is the event that should trigger this change and based on what criteria?

In that case it would be:

```
eventThatShouldTrigger
  .withLatestFrom(criteria)
```

`criteria` could then be a `combineLatest` of what we want to base the empty state on?